### ISSUE TITLE
Prevent scrolling on refresh/render

### ISSUE THREAD

**Issue by jemag** [nvim-neo-tree/neo-tree.nvim#781]
Currently it seems that refreshing neo-tree/re-render will scroll within the neo-tree view with a very large tree, see:

https://user-images.githubusercontent.com/7985687/222311811-3e663541-9eb4-48a7-ac83-1896563161d5.mp4

This is a bit jarring and throws the user off a smooth workflow. 

Would it be possible to prevent that scrolling from happening (basically cursor and buffer stays exactly the same)?

**Comment by miversen33** [comment]
Fwiw, refresh is actually fetching the file structure and rerendering it. 

I do agree that saving and restoring the cursor position would be ideal and should be easy enough to do.

Basically, we would need to get the current node in focus before the rerender (the node is part of the `state object` and may already be in scope of the rerender. If its not, you can get it via `state.tree:get_node()`)

Then perform the refresh/rerender. Lastly, call [focus_node](https://github.com/nvim-neo-tree/neo-tree.nvim/blob/v2.x/lua/neo-tree/ui/renderer.lua#L422-L515) on the node that we captured before render. 

Granted, this may still cause the contents of the window to shift (I don't know that there is a great way to prevent that as the rerender may add/remove items from the tree causing a shift), but the cursor should still be on the same object.

I don't remember what the code under refresh looks like, but this is the gist (as I remember it) of it.

I've tagged this as a "good first issue" as its relatively contained and should be a small change to perform :)
`

**Comment by jemag** [comment]
I guess its hard to see in my original video with the screenkey capture, but the cursor does not move position, there is no need to restore it.

It is simply scrolling through the buffer, even if no content is added/removed. I am not sure of the optimal way, but I am guessing you would essentially need to store the buffer state/which lines are visible and restore after render.

**Comment by jemag** [comment]
To clarify: basically if neo-tree is currently rendered in a grid, the text that the cursor is currently on, should remain at the same grid position (same row,col or x,y)
Currently the following starting point:
![image](https://user-images.githubusercontent.com/7985687/222920976-4693308d-92a9-45d1-898f-750c1f6da675.png)
can lead to this after render:
![image](https://user-images.githubusercontent.com/7985687/222921031-7c92b72c-eaeb-4043-92c7-d3c8bb4a6139.png)

Basically I think it would just stay from:
![image](https://user-images.githubusercontent.com/7985687/222920976-4693308d-92a9-45d1-898f-750c1f6da675.png)
to 
![image](https://user-images.githubusercontent.com/7985687/222920976-4693308d-92a9-45d1-898f-750c1f6da675.png)
no matter the surrounding text

For example:

https://user-images.githubusercontent.com/7985687/222921174-39252c9c-bd94-44fd-bb20-e903d8b24471.mp4

**Comment by ghostbuster91** [comment]
> I've tagged this as a "good first issue" as its relatively contained and should be a small change to perform :) 

I can give it a try in upcoming days if no one is working on it atm.

**Comment by pysan3** [comment]
Closing with the above PRs.
